### PR TITLE
update import_csv and add test coverage for it

### PR DESCRIPTION
**Pull request by andy-d**

### CODE REVIEW

File: import_csv.py [hunk 1]
```diff
@@ -15,179 +15,135 @@
...
+    '\x95', '').encode('ascii', 'replace')
 
 
-def import_sites(input_csv="../2012_ROOMS_site_info_sample.csv"):
-  """Change input_csv to actual input file - the default is test data."""
+def get_program(year):
+  assert isinstance(year, int)
+  nrd_type, created = ndb_models.ProgramType.get_or_create("NRD")
+  assert not created
+  program, created = ndb_models.Program.get_or_create(nrd_type.key, int(year))
```
Comment (babybunny) [inline import_csv.py]: minor: already asserted year is int

File: import_csv.py [hunk 2]
```diff
@@ -15,179 +15,135 @@
...
+  return program
+
+
+def import_sites(input_csv, nrd_program_year):
+  """
+  input_csv is a path like "../2012_ROOMS_site_info_sample.csv"
+
+  Change input_csv to actual input file - the default is test data.
```
Comment (babybunny) [inline import_csv.py]: no default anymore.

File: import_csv.py [hunk 3]
```diff
@@ -15,179 +15,135 @@
...
-    # Because Python 2.x csv module only reads ascii.
-    def clean_s(k):
-      return s[k].replace('\n', ' ').replace('\xe2', "'").replace('\x80', "'").replace('\x99', '').replace('\xc3',
-                                                                                                           '').replace(
-        '\x95', '').replace('\xb1', '').encode('ascii', 'replace')
-
-    site.name = clean_s("Repair Application: Applicant's Name")
-    site.street_number = clean_s("Street Address")
+      site = ndb_models.NewSite(number=number)
+    site.program = get_program(nrd_program_year).name
```
Comment (babybunny) [inline import_csv.py]: don't you need to set program_key?

File: import_csv.py [hunk 4]
```diff
@@ -15,179 +15,135 @@
...
+    budget = s.get("Budgeted Cost in Campaign", "$0").strip("$").replace(",", "") or '0'
```
Comment (babybunny) [inline import_csv.py]: way better, thanks

File: import_csv.py [hunk 5]
```diff
@@ -15,179 +15,135 @@
...
     logging.info('put site %s', number)
 
 
-def import_captains(input_csv="../2012_ROOMS_Captain_email_sample.csv"):
-  """Change input_csv to actual input file - the default is test data."""
+def import_captains(input_csv):
+  """
+  input_csv is a path like "../2012_ROOMS_site_info_sample.csv"
+
+  Change input_csv to actual input file - the default is test data.
```
Comment (babybunny) [inline import_csv.py]: no default here

File: import_csv.py [hunk 6]
```diff
@@ -15,179 +15,135 @@
...
       if captain:
         logging.info('got captain from key %s', key)
     if not captain:
-      captain = models.Captain.all().filter('rooms_id =', rooms_id).get()
+      captain = ndb_models.Captain.query(ndb_models.Captain.rooms_id == rooms_id).get()
       if captain:
         logging.info('got captain from rooms_id %s', rooms_id)
     if not captain:
-      captain = models.Captain.all().filter('email =', email).get()
+      captain = ndb_models.Captain.query().filter(ndb_models.Captain.email == email).get()
```
Comment (babybunny) [inline import_csv.py]: Here and in other places, can put  the filter param directly into query(). 

Like 
ndb_models.Captain.query(ndb_models.Captain.email == email)

see https://cloud.google.com/appengine/docs/standard/python/ndb/queries

File: test/test_import_csv.py [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    site = ndb_models.NewSite.query().filter(ndb_models.NewSite.number == "18010SAM").get()  # type: ndb_models.NewSite
+    self.assertTrue(site)
+    site_captains = list(site.sitecaptain_set)
+    self.assertEqual(len(site_captains), 2)
+    for site_captain in site_captains:
+      ctype = site_captain.type
+      captain = site_captain.captain.get()
+      self.assertIn(ctype, ("Construction", "Volunteer"))
+      if ctype == "Construction":
+        self.assertEqual(captain.email, 'mike@nibbi.com')
```
Comment (babybunny) [inline test/test_import_csv.py]: Let's avoid putting real email addresses in github.

File: test/test_import_csv.py [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+      else:
+        self.assertEqual(captain.email, 'bobn@nibbi.com')
+
+
+  SITES_DATA = """\
+Announcement Subject,Announcement Body,Site ID,Budgeted Cost in Campaign,Repair Application: Applicant's Name,Applicant Home Phone,Applicant Mobile Phone,Applicant Work Phone,Recipient's Street Address,Recipient's City,Recipient's Zip Code,Jurisdiction,Sponsor,Repair Application: RRP Test Results,Photos Link
+subject of announcement,"announcement body, has some commas",18001DAL,"$10,000",Man Shu Lau,,(415) 350-7499,,221 South Hill Court,Daly City,94014,Daly City,Woodlawn Foundation & St. Andrews Church (volunteers),"Not tested - Not required to test, assume no lead",https://flic.kr/s/aHsksxU9or
```
Comment (babybunny) [inline test/test_import_csv.py]: also here, real data is not appropriate. let's invent some made-up world based on Tiago's imagination.

**Review by babybunny — CHANGES_REQUESTED** [review]

**Comment by andy-d** [comment]
Ack I’ll sanitize test data and update based on feedback thanks

**Comment by andy-d** [comment]
Removed the previous commit to avoid private data leakage

**Pull request closed without merging**